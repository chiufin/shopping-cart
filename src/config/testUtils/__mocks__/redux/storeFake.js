const storeFake = state => ({
  default: () => {},
  subscribe: () => {},
  dispatch: () => {},
  getState: () => ({ ...state }),
})

export default storeFake
