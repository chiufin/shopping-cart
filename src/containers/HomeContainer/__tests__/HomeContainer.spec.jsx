import React from 'react'
import { shallow, mount } from 'enzyme'
import { Provider } from 'react-redux'
import { MemoryRouter } from 'react-router'
import ConnectedHomeContainer, { HomeContainer } from '../HomeContainer'
import storeFake from '../../../config/testUtils/__mocks__/redux/storeFake'

jest.mock('./HomeContainer.scss', () => ({
  'products-container': 'products-container',
}))

const defaultProps = {
  loadProductsData: jest.fn(),
  products: [
    {
      productId: '1234',
      price: '499.00',
      title: 'Test Product',
      image: 'https://media.4rgos.it/s/Argos/9805244_R_SET?$Web$&w=110&h=130',
    },
  ],
}

const defaultStoreProps = {
  app: {
    products: [...defaultProps.products],
  },
}

const store = storeFake({
  ...defaultStoreProps,
})

describe('HomeContainer', () => {
  it('Renders correctly', () => {
    const shallowHomeContainer = shallow(<HomeContainer {...defaultProps} />)
    expect(shallowHomeContainer).toHaveLength(1)
    expect(shallowHomeContainer).toMatchSnapshot()
  })

  it('Renders the wrapped component', () => {
    const mountedHomeContainer = mount(<Provider store={store}><MemoryRouter><ConnectedHomeContainer /></MemoryRouter></Provider>)
    expect(mountedHomeContainer).toHaveLength(1)
  })
})
