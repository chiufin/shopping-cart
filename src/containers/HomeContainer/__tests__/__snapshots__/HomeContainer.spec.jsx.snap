// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeContainer Renders correctly 1`] = `
<Fragment>
  <h1>
    Products
  </h1>
  <div
    className="products-container"
  >
    <Product
      key="1234"
      product={
        Object {
          "image": "https://media.4rgos.it/s/Argos/9805244_R_SET?$Web$&w=110&h=130",
          "price": "499.00",
          "productId": "1234",
          "title": "Test Product",
        }
      }
    />
  </div>
</Fragment>
`;
